prov/psm2/src/psmx2_cq.c

```diff
@@ -769,9 +769,12 @@ psmx2_mq_status_copy(struct psm2_mq_req_user *req, void *status_array, int entry
 			op_context = sendv_rep->user_context;
 			buf = sendv_rep->buf;
 			flags |= sendv_rep->comp_flag;
+			data = PSMX2_GET_CQDATA(sendv_rep->tag);
+			if (PSMX2_HAS_IMM(PSMX2_GET_FLAGS(sendv_rep->tag)))
+				flags |= FI_REMOTE_CQ_DATA;
 			err = psmx2_cq_rx_complete(
 					status_data->poll_cq, ep->recv_cq, ep->av,
-					req, op_context, buf, flags, 0,
+					req, op_context, buf, flags, data,
```

Comment:
On the send side, in both `psmx2_sendv_generic()` and `psmx2_tagged_sendv_generic()`, when `req->iov_protocol == PSMX2_IOV_PROTO_MULTI`, the send-tag is constructed with neither scalable endpoint ID nor CQ data and the  `PSMX2_IMM_BIT` is not set in this case. I don't know if there's anything the receiver has to do in this case.